packages/core/src/internal/utils/validators.ts

```diff
@@ -119,8 +119,16 @@ export function validateTopK(topK: number) {
 }
 
 export function validateTtlMinutes(ttlMinutes: number) {
-  if (ttlMinutes < 0) {
-    throw new InvalidArgumentError('ttlMinutes must be positive');
+  if (ttlMinutes < 0 || !Number.isInteger(ttlMinutes)) {
+    throw new InvalidArgumentError('ttlMinutes must be a positive integer');
+  }
+}
+
+export function validateTtl(ttl: number) {
+  if (ttl < 0 || !Number.isInteger(ttl)) {
```

Comment:
Should we use `Number.isSafeInteger` instead? I am not sure if a `ttl` beyond Int.MAX_VALUE is acceptable and if not `isSafeInteger` may be a better check. cc @cprice404